normandy/recipes/exports.py

```diff
@@ -0,0 +1,207 @@
+import logging
+
+import kinto_http
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+
+
+logger = logging.getLogger(__name__)
+
+
+APPROVE_CHANGES_FLAG = {"status": "to-sign"}
+
+
+def recipe_as_record(recipe):
+    """
+    Transform a recipe to a dict with the minimum amount of fields needed for clients
+    to verify and execute recipes.
+
+    :param recipe: a recipe ready to be exported.
+    :returns: a dict to be posted on Remote Settings.
+    """
+    from normandy.recipes.api.v1.serializers import (
+        MinimalRecipeSerializer,
+    )  # avoid circular imports
+
+    serializer = MinimalRecipeSerializer(recipe)
+    record = serializer.data
+    record["id"] = str(recipe.id)
+    return record
+
+
+class RemoteSettings:
+    """
+    Interacts with a RemoteSettings service.
+
+    Basically, a recipe becomes a record in the dedicated collection on Remote Settings.
+    When it is disabled, the associated record is deleted.
+
+    Since Normandy already has the required approval/signoff features, the integration
+    bypasses the one of Remote Settings (leveraging a specific server configuration for this
+    particular collection).
+
+    .. notes::
+
+        Remote Settings signoff workflow relies on several buckets (see kinto-signer API).
+
+        The `main-workspace` is only readable and writable by authorized accounts.
+        The `main` bucket is read-only, but publicly readable. The Remote Settings
+        clients pull data from there.
+
+        Since the review step is disabled for Normandy, publishing data is done in two steps:
+
+        1. Create, update or delete records in the `main-workspace` bucket
+        2. Approve the changes by flipping the ``status`` field to ``to-sign``
+           in the collection metadata
+        3. The server will sign and publish the new data to the `main` bucket.
+
+    """
+
+    MAIN_BUCKET_ID = "main"
+    """The name of the publicly readable bucket of Remote Settings where the approved
+    records are published.
+    """
+
+    def __init__(self):
+        self.collection_id = str(settings.REMOTE_SETTINGS_COLLECTION_ID)
+
+        # Kinto is the underlying implementation of Remote Settings. The client
+        # is basically a tiny abstraction on top of the requests library.
+        self.client = (
+            kinto_http.Client(
+                server_url=str(settings.REMOTE_SETTINGS_URL),
+                auth=(
+                    str(settings.REMOTE_SETTINGS_USERNAME),
+                    str(settings.REMOTE_SETTINGS_PASSWORD),
+                ),
+                bucket=str(settings.REMOTE_SETTINGS_BUCKET_ID),
```

Comment:
Why all these `str()` here? Django-configurations makes sure they have the right types.